### DISCUSSION TITLE
Performance testing of convolutions on Radeon VII: all solvers vs deprecated solvers disabled.

### DISCUSSION THREAD

**Discussion by atamazov** [ROCmSoftwarePlatform/MIOpen#2120] [General]
An investigation of an imaginary case where [several old OpenCL-based solvers](https://github.com/ROCmSoftwarePlatform/MIOpen/issues/2080) are turned off.

### Prerequisites
- gfx906/60
- ROCm 5.4.1
- MIOpen release build
  - DEV=Off to have binary cache
  - COMgr and HIPRTC are disabled to enable parallel builds (for performance of tuning)
- The configurations under test are fully tuned locally:
  - MIOPEN_FIND_ENFORCE=3 (this fills the omissions in perf-db, if any)
  - The binary kernel cache is filled with necessary kernels
  - The User Find-Db is also updated.

### Results
- All solvers enabled vs deprecated solvers disabled
  - FP32 GPU time 0.85 (average perf.gain), 0.51 (sum of times)
  - FP16 GPU time 0.81 (average perf.gain), 0.31 (sum of times)
- The same but iGEMM ASM solvers constantly disabled (no MISA-generated kernels)
  - FP32 GPU time 0.84 (average perf.gain), 0.48 (sum of times)
  - FP16 GPU time 0.81 (average perf.gain), 0.32 (sum of times)

The most affected convolution is WrW
- fp32 0.76 (avg) 0.32 (sum)
- fp16 0.58 (avg) 0.18 (sum)

The least affected convolution is Forward -- its performance drop is less than 5%.

### Bottom line
Most likely, inference won't be affected, but I am expecting ~2X perf drop for FP32 training and possibly even worse for FP16 training.

----
CC @junliume @JehandadKhan
- https://github.com/ROCmSoftwarePlatform/MIOpen/labels/performance
